src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1455,13 +1456,18 @@
       ;; no new CRLs found for this issuer, keep it
       crl)))
 
+(schema/defn get-auth-key-id
+  [crl :- X509CRL]
+  (if-let [key-id (utils/get-extension-value crl utils/authority-key-identifier-oid)]
+    key-id
+    (throw (IllegalArgumentException. ^String (i18n/trs "One or more submitted CRLs do not have an authority key identifier.")))))
+
 (schema/defn ^:always-validate update-crls
   "Given a collection of CRLs, update the CRL chain and confirm that
   all CRLs are currently valid."
   [incoming-crls :- [X509CRL]
    crl-path :- schema/Str
    cert-chain-path :- schema/Str]
-  (log/info (i18n/trs "Updating CRL at {0}" crl-path))
```

Comment:
I found it very misleading when I was triggering error cases with invalid input to see something like "Updating CRL...No valid CRLs." My first thought was like "Well you weren't updating it then, were you?" I think the issue that is that it was being issued before we were done validating our inputs even. So even though we knew we weren't going to even attempt to update (which is a little different than an update that errors), we still said we were.

Reply:
hmm. I guess in my mind, validating inputs is still part of the attempt to update. but if you think it would be clear to a user how they encountered this error, then I think it's fine. another option would be to put this back here but then put another message at the end to say the update was successful? but maybe that's overkill. ¯\_(ツ)_/¯

Reply:
Let me know what you think of this! I liked the effect in my testing.
```
2021-10-08 15:02:48,503 INFO  [qtp834154798-71] [p.p.certificate-authority] Processing update to CRL at /Users/aileen/.puppetlabs/etc/puppet/ssl/ca/ca_crl.pem
2021-10-08 15:02:48,563 ERROR [qtp834154798-71] [p.s.c.certificate-authority-core] One or more submitted CRLs do not have an authority key identifier.
2021-10-08 15:03:06,666 INFO  [qtp834154798-71] [p.p.certificate-authority] Processing update to CRL at /Users/aileen/.puppetlabs/etc/puppet/ssl/ca/ca_crl.pem
2021-10-08 15:03:06,678 ERROR [qtp834154798-71] [p.s.c.certificate-authority-core] Could not determine newest CRL.
2021-10-08 15:06:43,849 INFO  [qtp834154798-74] [p.p.certificate-authority] Processing update to CRL at /Users/aileen/.puppetlabs/etc/puppet/ssl/ca/ca_crl.pem
2021-10-08 15:06:43,863 INFO  [qtp834154798-74] [p.p.certificate-authority] Updating CRL for issuer EMAILADDRESS=maggie@puppet.com, CN=maggie, OU=Eng, O=Puppet, L=Portland, ST=OR, C=US.
2021-10-08 15:06:43,895 INFO  [qtp834154798-74] [p.p.certificate-authority] Successfully updated CRL at /Users/aileen/.puppetlabs/etc/puppet/ssl/ca/ca_crl.pem
```